lib/tpm2_identity_util.c

```diff
@@ -116,6 +118,15 @@ static bool encrypt_seed_with_tpm2_rsa_public_key(TPM2B_DIGEST *protection_seed,
 
     TPMI_ALG_HASH parent_name_alg = parent_pub->publicArea.nameAlg;
 
+    /*
+     * RSA Secret Sharing uses a randomly generated seed (Part 1, B.10.3).
+     */
+    protection_seed->size = tpm2_alg_util_get_hash_size(parent_name_alg);
+    if (RAND_bytes(protection_seed->buffer, protection_seed->size) <= 0) {
```

Comment:
we try not to hide conditionals. please assign the function output to an rc variable

Reply:
Done.